docs/gsoc.rst

```diff
@@ -164,16 +140,33 @@ We should rebuild how we handle redirects across a number of cases:
 
 There will also be a good number of things that spawn from this, including version aliases and other related concepts, if this task doesn't take the whole summer.
 
+Autobuild docs for Pull Requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+It would be great to automatically build docs for Pull Requests in GitHub repos that our users have.
+Currently we don't support this,
+and it's one of our most requested features.
+
+This would include:
+
+* Modeling Pull Requests as a type of version alongside Tags and Branches
+* Modifying how we upload HTML docs to store them in a place like S3 for long term storage
```

Comment:
Long term storage for docs from a PR?

Reply:
Yea, we really don't want them on our web servers which are already space limited.